### PR TITLE
cmake: building for generic openwrt targets

### PR DESCRIPTION
**Pull request by dob71**
- Removed remaining refences to turris-omnia
- Added ability to customize interface names and BWL_TYPE for generic
  openwrt targets using the following environment variables:
    PRPLMESH_WLAN1_IFACE (if not set default is wlan0)
    PRPLMESH_WLAN2_IFACE (if not set default is wlan1)
    PRPLMESH_BH_WIRE_IFACE (if not set default is eth1)
    PRPLMESH_BWL_TYPE (if not set default is NL80211)

Note: it's no longer needed to set PRPLMESH_PLATFORM_TYPE for turris-omnia
      or any other openwrt target except ugw, but the changes are backward
      compatible and the build should still work if the above environment
      variable is set.

### CODE REVIEW

**Review by tomereli — CHANGES_REQUESTED** [review]
Why is it preferred to have these as environment variables instead of simple cmake variables? I'm asking since in the OpenWrt Makefile we will now need to define environment variables instead of simple cmake variables which I'd prefer to avoid.
This should be the usage in the OpenWrt feed:

```
CMAKE_OPTIONS += \
-DBEEROCKS_WLAN1_IFACE=wlan0 \
-DBEEROCKS_WLAN2_IFACE=wlan2
-DBWL_TYPE=NL89211

```

So I don't see why we need all these environment variables. 

These are used for out of tree build with tool chain file, which uses env vars and sets cmake vars.

**Review by rmelotte — CHANGES_REQUESTED** [review]
I agree with what @tomereli said. It should be possible to use cmake variables directly?
Also (I might be wrong) I don't think new variables names are required, you could reuse the existing ones.

We need a way to change them using the "-D" option though.
You could either:
- use set(...CACHE) to cache them (making it possible to change them using "-D").
- check if they are already defined:
```
if (NOT DEFINED BEEROCKS_WLAN1_IFACE)
  set(BEEROCKS_WLAN1_IFACE "wlan2")
endif()
```
I don't know if there is any specific reason why they were not using CACHE already, since it's already the case for the target platform and so on. Could anyone else comment on that?

Also since this PR is about making it possible to change the interface names, I suggest you also grep for the remaining places where they are hard-coded (grep for `wlan0` for example: it's still present in `prplmesh_utils.sh.in`) and change them to use the cmake variables (only for the places which are already generated by cmake, so only the `.in` files).

**Comment by arnout** [comment]
> I don't know if there is any specific reason why they were not using CACHE already, since it's already the case for the target platform and so on. Could anyone else comment on that?

Because the original philosophy was that these variables were "internal" variables that are set based on PLATFORM. We are now changing our mind and saying that the platform's build system should specify those details and there should not be a PLATFORM at all.

Note that specifically for the interface names, we don't need to bother too much with the details since they're going to be removed entirely by #574.

> Also since this PR is about making it possible to change the interface names, I suggest you also grep for the remaining places where they are hard-coded (grep for `wlan0` for example: it's still present in `prplmesh_utils.sh.in`) and change them to use the cmake variables (only for the places which are already generated by cmake, so only the `.in` files).

:+1:

File: CMakeLists.txt
Comment (arnout) [inline CMakeLists.txt]: Since all of this will anyway be deprecated by #574 I think we should go for the simple solution:

```
... ugw stuff ...
else()
    set(BEEROCKS_WLAN2_IFACE "wlan1")
endif()
```

File: common/beerocks/bwl/CMakeLists.txt
Comment (arnout) [inline common/beerocks/bwl/CMakeLists.txt]: Here, I think the existing code already works, so this change is not needed.

Long term, we want BWL_TYPE to be a cache variable and only set it based on PLATFORM if it is not defined already. Longer term, we'll want to remove PLATFORM entirely.

File: framework/platform/bpl/CMakeLists.txt
Comment (arnout) [inline framework/platform/bpl/CMakeLists.txt]: The comment "Platform defintiions" was about the add_definitions statement, so this comment is not correct. Just remove it.

File: tools/cmake/toolchain/openwrt.cmake
Comment (arnout) [inline tools/cmake/toolchain/openwrt.cmake]: As you mention in the commit message, after this change only "ugw" is used in TARGET_PLATFORM_TYPE. So there's no need to define it as generic, you can just as well leave it empty. So I don't think this change is needed.

**Review by arnout — CHANGES_REQUESTED** [review]

File: CMakeLists.txt
Comment (dob71) [reply CMakeLists.txt]: Do you mean revert it to what it was, just set BEEROCKS_WLAN2_IFACE to "wlan1"?
At minimum I need to be able to set wired backhaul to eth0 fro Gl.Inet and  keep it eth1 for Turris. I don't mind removing all the wlan# interface stuff if it will no longer be needed soon.

File: common/beerocks/bwl/CMakeLists.txt
Comment (dob71) [reply common/beerocks/bwl/CMakeLists.txt]: Ok, will remove

File: framework/platform/bpl/CMakeLists.txt
Comment (dob71) [reply framework/platform/bpl/CMakeLists.txt]: Ok

File: tools/cmake/toolchain/openwrt.cmake
Comment (dob71) [reply tools/cmake/toolchain/openwrt.cmake]: I was thinking that it might become useful for printing a message, but sure can remove.

**Review by dob71 — COMMENTED** [review]
@tomereli The changes were made to allow compiling prplMesh separately outside of the openwrt tree per the instructions on the WiKi pages (following the same pattern that was used for Turris): 
https://github.com/prplfoundation/prplMesh/wiki/prplMesh-on-Gl.Inet-B1300#cross-compile-prplmesh-for-glinet-b1300
The environment vars were just easier to use in that case.

File: tools/cmake/toolchain/openwrt.cmake
Comment (arnout) [inline tools/cmake/toolchain/openwrt.cmake]: Why is this condition needed? If TARGET_PLATFORM_TYPE is not defined in the environment, it will just be defined to empty string and that's OK, no?

Comment (dob71) [reply tools/cmake/toolchain/openwrt.cmake]: I don't know cmake, so trying to stay safe, I'll try what happens if it's left undefined and if it works  remove the condition.

**Review by arnout — APPROVED** [review]
I squashed everything into a single commit and updated the commit message a bit.

**Review by tomereli — APPROVED** [review]

**Review by rmelotte — CHANGES_REQUESTED** [review]
Typo in the commit message: `PRPLMESH_PLATFORM_TYPE` should be `TARGET_PLATFORM_TYPE`.
Otherwise, LGTM.

**Comment by arnout** [comment]
> `PRPLMESH_PLATFORM_TYPE`

Fixed.

If you only have such a minor comment, please approve the PR already so it can be merged without your re-review. You can use the new "don't merge" tag to indicate that it still needs some fixup before merge (anybody can remove that tag, while admin privilege is required to override a reviewer's "changes requested").

**Comment by rmelotte** [comment]
> If you only have such a minor comment, please approve the PR already so it can be merged without your re-review. You can use the new "don't merge" tag to indicate that it still needs some fixup before merge (anybody can remove that tag, while admin privilege is required to override a reviewer's "changes requested").

Alright, will do. I was thinking maybe anyone could bypass/reset my "change request" because I'm not in the code owners.

**Review by rmelotte — APPROVED** [review]

**Pull request merged**
